mrp_production_batch/models/mrp_production.py

```diff
@@ -0,0 +1,292 @@
+        if mrp_productions.production_batch_id:
+            raise ValidationError(
+                _("Some of the selected productions already have a batch")
+            )
+        if {"done", "cancel"} & set(mrp_productions.mapped("state")):
+            raise ValidationError(
+                _("Some of the selected productions are already done or cancelled")
+            )
+        if len(mrp_productions.picking_type_id) > 1:
+            raise ValidationError(
+                _("Some of the selected productions have different operation types")
+            )
+        if len(mrp_productions.picking_type_id.warehouse_id) > 1:
+            raise ValidationError(
+                _("Some of the selected productions have different warehouses")
+            )
+        if len(mrp_productions.picking_type_id) > 1:
+            raise ValidationError(
+                _(
+                    "Some of the selected productions have different picking types:%s"
+                    % mrp_productions.picking_type_id.mapped("name")
+                )
+            )
+
+    def mrp_production_batch_create_wizard_action(self):
+        if (
+            "active_model" not in self.env.context
+            or "active_ids" not in self.env.context
+        ):
+            raise ValidationError(
+                _(
+                    "An unexpected error has occurred. Please try again, and "
+                    "if the problem persists, contact technical support."
+                )
+            )
+
+        model = self.env.context.get("active_model")
+        mrp_production_ids = self.env[model].browse(self.env.context.get("active_ids"))
+        self._check_production_to_batch_consistency(mrp_production_ids)
+        view_form = self.env.ref(
+            "mrp_production_batch.mrp_production_batch_wizard_view_form"
+        )
+        res_id = self.env["mrp.production.batch.wizard"].create(
+            {
+                "warehouse_id": mrp_production_ids.mapped("picking_type_id")
+                .mapped("warehouse_id")
+                .id,
+            }
+        )
+        res = {
+            "name": _("MRP production Batch"),
+            "view_mode": "form",
+            "res_model": "mrp.production.batch.wizard",
+            "res_id": res_id.id,
+            "target": "new",
+            "views": [(view_form.id, "form")],
+            "view_id": view_form.id,
+            "type": "ir.actions.act_window",
+            "context": dict(self.env.context, active_ids=self.ids),
+        }
+        return res
+
+    def _action_generate_consumption_wizard(self, consumption_issues):
+        if not self.env.context.get("mrp_production_batch_create"):
+            return super()._action_generate_consumption_wizard(consumption_issues)
+        bom_products = set(self.bom_id.bom_line_ids.product_id.ids)
+        production_products = set(self.move_raw_ids.product_id.ids)
+        if bom_products != production_products:
```

Comment:
A recordset is a set, you can use them directly
```suggestion
        if self.bom_id.bom_line_ids.product_id != self.move_raw_ids.product_id:
```